java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestGenerationUtils.java

```diff
@@ -0,0 +1,407 @@
+            return null;
+        }
+
+        final ITypeBinding binding = ((TypeDeclaration) coveringNode).resolveBinding();
+        if (binding == null) {
+            JUnitPlugin.logError("Failed to resolve type binding from " + unit.getElementName());
+            return null;
+        }
+
+        final IJavaProject javaProject = unit.getJavaProject();
+        if (javaProject == null) {
+            JUnitPlugin.logError("Cannot get Java project from " + unit.getElementName());
+            return null;
+        }
+
+        for (final IClasspathEntry entry : javaProject.readRawClasspath()) {
+            if (entry.getPath().isPrefixOf(unit.getPath())) {
+                if (ProjectTestUtils.isTestEntry(entry)) {
+                    return generateTestsFromTest(unit, root, (TypeDeclaration) coveringNode, binding, cursorPosition);
+                } else {
+                    generateTestsFromSource();
+                }
+            }
+        }
+
+        return null;
+    }
+
+    private static void generateTestsFromSource() {
+        // TODO: unimplement
+    }
+
+    private static WorkspaceEdit generateTestsFromTest(ICompilationUnit unit, CompilationUnit root,
+            TypeDeclaration typeNode, ITypeBinding typeBinding, int cursorPosition)
+            throws MalformedTreeException, CoreException {
+        final Set<TestKind> availableFrameworks = getTestKindFromFile(unit);
+        if (availableFrameworks.size() == 0) {
+            availableFrameworks.addAll(TestKindProvider.getTestKindsFromCache(unit.getJavaProject()));
+        }
+
+        final TestKind testKind = determineTestFramework(availableFrameworks);
+        if (testKind == null) {
+            return null;
+        }
+
+        final List<String> lifecycleAnnotations = getLifecycleAnnotations(testKind);
+        final List<String> methodsToGenerate = determineMethodsToGenerate(lifecycleAnnotations);
+
+        if (methodsToGenerate == null || methodsToGenerate.size() == 0) {
+            return null;
+        }
+
+        IJavaElement insertPosition = null; // Insert to the last by default.
+        try {
+            insertPosition = CodeGenerationUtils.findInsertElement(
+                        (IType) typeBinding.getJavaElement(), cursorPosition);
+        } catch (Throwable e) {
+            // ignore if the upstream does not support insert position preference.
+        }
+
+        final TextEdit textEdit = createTextEdit(testKind, methodsToGenerate, root, typeNode,
+                typeBinding, insertPosition);
+        return SourceAssistProcessor.convertToWorkspaceEdit(unit, textEdit);
+    }
+
+    private static List<String> determineMethodsToGenerate(List<String> lifecycleAnnotations) {
+        final List<String> methodList = lifecycleAnnotations.stream()
+                .map(annotation -> "@" + annotation.substring(annotation.lastIndexOf(".") + 1) + " Method")
+                .collect(Collectors.toList());
```

Comment:
If the helper `getLifeCycleAnnotations()` just returns the simple annotation names without prefix, you could use lambda `annotation -> "@" + annotation + " Method"` directly.

so suggest to refactor  `getLifeCycleAnnotations()` to return the simple names instead of qualified names.